File: Core CS/Algorithms/Implementation/Breaking the Records/breakingTheRecords.cpp
#include <bits/stdc++.h>
using namespace std;

vector < int > getRecord(vector < int > s){
    int maxc = 0,
        minc = 0,
        max = s[0],
        min = s[0];
        
    for(int i=1;i<s.size();i++){
        if(max < s[i]){
            max = s[i];
            maxc++;
        }
        else if(min > s[i]){
            min = s[i];
            minc++;
        }
    }
    vector<int> r{maxc,minc};
    return r;
}

int main() {
    int n;
    cin >> n;
    vector<int> s(n);
    for(int s_i = 0; s_i < n; s_i++){
       cin >> s[s_i];
    }
    vector < int > result = getRecord(s);
    string separator = "", delimiter = " ";
    for(auto val: result) {
        cout<<separator<<val;
        separator = delimiter;
    }
    cout<<endl;
    return 0;
}

File: Core CS/Algorithms/Implementation/Breaking the Records/breakingTheRecords.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function breakingRecords(s) {
    let maxc = 0,
        minc = 0,
        max = s[0],
        min = s[0];
    
    for(let i=1;i<s.length;i++){
        if(max < s[i]){
            max = s[i];
            maxc++;
        }
        else if(min > s[i]){
            min = s[i];
            minc++;
        }
    }
    return new Array(maxc, minc);

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    const scores = readLine().split(' ').map(scoresTemp => parseInt(scoresTemp, 10));

    const result = breakingRecords(scores);

    ws.write(result.join(' ') + '\n');

    ws.end();
}
